docs/manual/fread_examples.rst

```diff
@@ -0,0 +1,490 @@
+
+
+Fread Examples
+=================
+
+Read Data
+----------
+
+Read text data directly::
+
+    from datatable import dt, fread
+
+    data = ('col1,col2,col3\n'
+            'a,b,1\n'
+            'a,b,2\n'
+            'c,d,3')
+
+    fread(data)
+
+    col1  col2  col3
+     a     b     1
+     a     b     2
+     a     b     3
+
+Read from text file (any file that has the `.read()` method)::
+
+    result = fread('iris.csv')
+    result.head(5)
+
+        sepal_length	sepal_width	petal_length	petal_width	species
+    0	5.1	            3.5	            1.4	            0.2	        setosa
+    1	4.9	            3	            1.4	            0.2	        setosa
+    2	4.7	            3.2	            1.3	            0.2	        setosa
+    3	4.6	            3.1	            1.5	            0.2	        setosa
+    4	5	            3.6	            1.4	            0.2	        setosa
+
+
+Read from url::
+
+    url = "https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv"
+    fread(url)
+
+Read from an archive. If there are multiple files, only the first will be read; you can specify the path to the specific file you are interested in::
+
+
+    fread("data.zip/mtcars.csv")
+
+Read from `.xlsx` files - `.xls` files are not supported::
+
+    fread("excel.xlsx")
+
+For excel files, you can specify the sheet to be read::
+
+    fread("excel.xlsx/Sheet1")
+
+Note: `xlrd <https://pypi.org/project/xlrd/>`_ must be installed to read in excel files.
+
+You can also read in data from the command line. Simply pass the command line statement to the `cmd` parameter::
+
+     #https://blog.jpalardy.com/posts/awk-tutorial-part-2/
+     #You specify the `cmd` parameter
+     #Here we filter data for the year 2015
+     fread(cmd = """cat netflix.tsv | awk 'NR==1; /^2015-/'""")
+
+The command line can be very handy with large data; you can do some of the preprocessing before reading in the data to `datatable`.
+
+Dealing with Unicode Data
+-------------------------
+
+`Fread` is pretty powerful and can automatically detect the encoding in a number of cases.::
+
+    data = ("num,name\n"
+            "1,alïce\n"
+            "2,bøb\n"
+            "3,cárol\n"
+            )
+
+    fread(data)
+
+              num       name
+        0	1	alïce
+        1	2	bøb
+        2	3	cárol
+
+You can also specify the encoding; `fread` will recode into `UTF-8` before reading.
+Check `here <https://docs.python.org/3/library/codecs.html#standard-encodings>`_ for Python's list of standard encodings.
+
+Detect Thousand Separator
+-------------------------
+
+Thousand separators are automatically detected::
```

Comment:
Actually we don't detect the thousand separator, we just assume it's `,`. 
Looks like this parameter is not even properly exposed to the user, though it exists internally.

I guess we could set it automatically based on `dec`.